crates/ruma-federation-api/src/authenticated_media.rs

```diff
@@ -0,0 +1,519 @@
+//! Authenticated endpoints for the content repository, according to [MSC3916].
+//!
+//! [MSC3916]: https://github.com/matrix-org/matrix-spec-proposals/pull/3916
+
+use ruma_common::http_headers::ContentDisposition;
+use serde::{Deserialize, Serialize};
+
+pub mod get_content;
+pub mod get_content_thumbnail;
+
+/// The `multipart/mixed` mime "essence".
+const MULTIPART_MIXED: &str = "multipart/mixed";
+/// The maximum number of headers to parse in a body part.
+const MAX_HEADERS_COUNT: usize = 32;
```

Comment:
No, we just have to pick one because of httparse's API, to avoid bad actors using up all the memory. It's the value used by multer and it seems reasonable.